homeassistant/components/nexia/climate.py

```diff
@@ -0,0 +1,524 @@
+        if system_status == SYSTEM_STATUS_COOL:
+            return CURRENT_HVAC_COOL
+        if system_status == SYSTEM_STATUS_HEAT:
+            return CURRENT_HVAC_HEAT
+        if system_status == SYSTEM_STATUS_IDLE:
+            return CURRENT_HVAC_IDLE
+        return "idle"
+
+    @property
+    def hvac_mode(self):
+        """Operation requested ie. heat, cool, idle."""
+        return self.mode
+
+    @property
+    def hvac_modes(self):
+        """List of HVAC available modes."""
+        return [
+            HVAC_MODE_OFF,
+            HVAC_MODE_AUTO,
+            HVAC_MODE_HEAT_COOL,
+            HVAC_MODE_HEAT,
+            HVAC_MODE_COOL,
+        ]
+
+    @property
+    def mode(self):
+        """Return current mode, as the user-visible name."""
+
+        mode = self._device.get_requested_mode()
+        hold = self._device.is_in_permanent_hold()
+
+        if mode == OPERATION_MODE_OFF:
+            return HVAC_MODE_OFF
+        if not hold and mode == OPERATION_MODE_AUTO:
+            return HVAC_MODE_AUTO
+        if mode == OPERATION_MODE_AUTO:
+            return HVAC_MODE_HEAT_COOL
+        if mode == OPERATION_MODE_HEAT:
+            return HVAC_MODE_HEAT
+        if mode == OPERATION_MODE_COOL:
+            return HVAC_MODE_COOL
+        raise KeyError(f"Unhandled mode: {mode}")
+
+    def set_temperature(self, **kwargs):
+        """Set target temperature."""
+        new_heat_temp = kwargs.get(ATTR_TARGET_TEMP_LOW, None)
+        new_cool_temp = kwargs.get(ATTR_TARGET_TEMP_HIGH, None)
+        set_temp = kwargs.get(ATTR_TEMPERATURE, None)
+
+        deadband = self._device.thermostat.get_deadband()
+        cur_cool_temp = self._device.get_cooling_setpoint()
+        cur_heat_temp = self._device.get_heating_setpoint()
+        (min_temp, max_temp) = self._device.thermostat.get_setpoint_limits()
+
+        # Check that we're not going to hit any minimum or maximum values
+        if new_heat_temp and new_heat_temp + deadband > max_temp:
+            new_heat_temp = max_temp - deadband
+        if new_cool_temp and new_cool_temp - deadband < min_temp:
+            new_cool_temp = min_temp + deadband
+
+        # Check that we're within the deadband range, fix it if we're not
+        if new_heat_temp and new_heat_temp != cur_heat_temp:
+            if new_cool_temp - new_heat_temp < deadband:
+                new_cool_temp = new_heat_temp + deadband
+        if new_cool_temp and new_cool_temp != cur_cool_temp:
+            if new_cool_temp - new_heat_temp < deadband:
+                new_heat_temp = new_cool_temp - deadband
+
+        self._device.set_heat_cool_temp(
+            heat_temperature=new_heat_temp,
+            cool_temperature=new_cool_temp,
+            set_temperature=set_temp,
+        )
+        self.schedule_update_ha_state()
+
+    @property
+    def is_aux_heat(self):
+        """Emergency heat state."""
+        return "on" if self._device.thermostat.is_emergency_heat_active() else "off"
```

Comment:
This should be a boolean.
```suggestion
        return self._device.thermostat.is_emergency_heat_active()
```

Reply:
Thanks .. Missed that in the refactor